lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -39,7 +39,19 @@ use virt_utils;
 
 our @EXPORT
   = qw(download_network_cfg prepare_network restore_standalone destroy_standalone restart_libvirtd restart_network restore_guests restore_network
-  destroy_vir_network restore_libvirt_default enable_libvirt_log ssh_setup upload_debug_log check_guest_status save_guest_ip test_network_interface hosts_backup hosts_restore);
+  destroy_vir_network restore_libvirt_default enable_libvirt_log ssh_setup upload_debug_log check_guest_status check_guest_module save_guest_ip test_network_interface hosts_backup hosts_restore);
+
+sub check_guest_module {
```

Comment:
Thanks for looking back into this! I find this routine still a bit misleading: It allows to pass the `module` argument, but will not do anything if the module is not `acpiphp`.
Since this is only used internally and only for `acpiphp`, why not remove the `module` argument?

    sub check_acpiphp_module {
        my ($guest, %args) = @_;
        my $module = "acpiphp";
        if ($guest =~ m/sles-?11/i) {
            my $status = script_run("ssh root\@$guest \"lsmod | grep $module\"");
            if ($status != 0) {
                script_run("ssh root\@$guest modprobe $module", 60);
                record_info('bsc#1167828 - need to load $module kernel module to $guest guest otherwise network interface hotplugging does not work');
            }
        }
    }

That's just a suggestion, since it seems to work and the routine is not exported I'm also OK with leaving it as it is.

Reply:
@grisu48 good point. 
Regarding check_acpiphp_module func, it just only work for sles11sp4 guest system due to bsc#1167828 for now, used with module argument. 
But, I had some ideas for the module argument, not just only for acpiphp (worked for sles11sp4 guest system). 
Refer to bsc#1167828, we'd better to check with the loaded kernel module under guest system before our virtual network (i.e hotpluging) tests, do not just only focus on acpiphp as the module argument the for sless11sp4 guest system (older sles version). 
So, keep on still using the module argument for check_acpiphp_module  func here.

Reply:
I agree with @guoxuguang let's reuse this!

Reply:
@pdostal Awesome, thank you so much for your support.